buildSrc/src/main/kotlin/CodegenTestCommon.kt

```diff
@@ -102,3 +107,135 @@ fun cargoCommands(properties: PropertyRetriever): List<Cargo> {
     }
     return ret
 }
+
+fun registerGenerateSmithyBuildTask(
+    rootProject: Project,
```

Comment:
optional: could change these to all be extension functions on project—
```kotlin
fun Project.registerGenerateSmithyBuildTask() { ... }
```

which would make them all usable as `project.registerGenerateSmithyBuildTask(<args>)` — might be slightly clearer intent in the build.gradle

Reply:
Excellent suggestion. Done in https://github.com/awslabs/smithy-rs/pull/1422/commits/0c969bad2f974a3a1dfee55f3293d173a4e3247c.